pkg/diag/validator/config_connector.go

```diff
@@ -0,0 +1,175 @@
+		// TODO: add resource logs
+		rs = append(rs, NewResourceFromObject(&r, Status(resourceStatus.result.Status), &resourceStatus.ae, nil))
+	}
+	return rs, nil
+}
+
+func getResourceStatus(res unstructured.Unstructured) *configConnectorResourceStatus {
+	status := &configConnectorResourceStatus{
+		name:      res.GetName(),
+		namespace: res.GetNamespace(),
+		ae: proto.ActionableErr{
+			ErrCode: proto.StatusCode_STATUSCHECK_SUCCESS,
+		},
+	}
+
+	// config connector resource statuses follow the Kubernetes kstatus so we use the attached kstatus library
+	// https://github.com/kubernetes-sigs/cli-utils/tree/master/pkg/kstatus#the-ready-condition
+	result, err := kstatus.Compute(&res)
+	if err != nil || result == nil {
+		status.result = kstatus.Result{Status: kstatus.UnknownStatus}
+		status.updateAE(proto.StatusCode_STATUSCHECK_UNKNOWN, "unable to check resource status")
+		return status
+	}
+	status.result = *result
+	switch result.Status {
+	case kstatus.CurrentStatus:
+		return status
+	case kstatus.InProgressStatus:
+		if result.Message == "" {
+			status.updateAE(proto.StatusCode_STATUSCHECK_CONFIG_CONNECTOR_IN_PROGRESS, result.Message)
+		} else {
+			// config connector status doesn't always correctly parse to failed, but shows InProgress with an error message
+			status.updateAE(proto.StatusCode_STATUSCHECK_CONFIG_CONNECTOR_FAILED, result.Message)
+		}
+
+	case kstatus.FailedStatus:
+		status.updateAE(proto.StatusCode_STATUSCHECK_CONFIG_CONNECTOR_FAILED, result.Message)
+	case kstatus.TerminatingStatus:
+		status.updateAE(proto.StatusCode_STATUSCHECK_CONFIG_CONNECTOR_TERMINATING, result.Message)
+	case kstatus.NotFoundStatus:
+		status.updateAE(proto.StatusCode_STATUSCHECK_CONFIG_CONNECTOR_NOT_FOUND, result.Message)
+	case kstatus.UnknownStatus:
+		status.updateAE(proto.StatusCode_STATUSCHECK_UNKNOWN, result.Message)
+	}
+	return status
+}
+
+func processResourceEvents(ctx context.Context, e corev1.EventInterface, res unstructured.Unstructured, rs *configConnectorResourceStatus) {
+	log.Entry(ctx).Debugf("Fetching events for config connector resource %q", res.GetName())
+	// Get pod events.
+	scheme := runtime.NewScheme()
+	scheme.AddKnownTypes(v1.SchemeGroupVersion, &res)
+	events, err := e.Search(scheme, &res)
+	if err != nil {
+		log.Entry(ctx).Debugf("Could not fetch events for resource %q: %v", res.GetName(), err)
+		return
+	}
+	// find the latest event.
+	var recentEvent *v1.Event
+	for _, e := range events.Items {
+		event := e.DeepCopy()
+		if recentEvent == nil || recentEvent.LastTimestamp.Before(&event.LastTimestamp) {
+			recentEvent = event
+		}
+	}
+	if recentEvent == nil || recentEvent.Type == v1.EventTypeNormal {
+		return
+	}
+	// TODO: Add unique error codes for reasons
+	rs.updateAE(
+		proto.StatusCode_STATUSCHECK_UNKNOWN_EVENT,
+		fmt.Sprintf("%s: %s", recentEvent.Reason, recentEvent.Message),
+	)
+}
+
+type configConnectorResourceStatus struct {
+	name      string
+	namespace string
```

Comment:
A name is only unique within a kind and namespace ([source](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names)).  We should include the kind?